evm/src/cpu/kernel/assembler.rs

```diff
@@ -132,13 +132,29 @@ fn expand_macro_call(
         args.len()
     );
 
+    let get_arg = |var| {
+        let param_index = _macro.get_param_index(var);
+        args[param_index].clone()
+    };
+
     let expanded_item = _macro
         .items
         .iter()
         .map(|item| {
             if let Item::Push(PushTarget::MacroVar(var)) = item {
-                let param_index = _macro.get_param_index(var);
-                Item::Push(args[param_index].clone())
+                Item::Push(get_arg(var))
+            } else if let Item::MacroCall(name, args) = item {
```

Comment:
When inlining a macro call, we were only looking for vars (to replace with our args) in PUSHs, not in nested macro calls.